lambdas/check-coi/src/main/java/uk/gov/di/ipv/core/checkcoi/CheckCoiHandler.java

```diff
@@ -84,11 +99,41 @@ public Map<String, Object> handleRequest(ProcessRequest request, Context context
             var clientOAuthSession =
                     clientOAuthSessionDetailsService.getClientOAuthSession(
                             ipvSession.getClientOAuthSessionId());
-            LogHelper.attachGovukSigninJourneyIdToLogs(
-                    clientOAuthSession.getGovukSigninJourneyId());
+            var govukSigninJourneyId = clientOAuthSession.getGovukSigninJourneyId();
+            LogHelper.attachGovukSigninJourneyIdToLogs(govukSigninJourneyId);
 
             var userId = clientOAuthSession.getUserId();
 
+            var auditEventUser =
```

Comment:
Can we put all this in a separate method? I think it's distinct from the main logic for this lambda.

Reply:
Do you mean the coiCheckType switch statement as well?